crates/solver/src/main.rs

```diff
@@ -517,15 +526,29 @@ async fn main() {
             .await
             .map_err(|err| tracing::error!("Couldn't fetch market makable token list: {}", err))
             .ok();
-    let solution_submitter = SolutionSubmitter {
-        web3: web3.clone(),
-        contract: settlement_contract.clone(),
-        gas_price_estimator: gas_price_estimator.clone(),
-        target_confirm_time: args.target_confirm_time,
-        max_confirm_time: args.max_submission_seconds,
-        retry_interval: args.submission_retry_interval_seconds,
-        gas_price_cap: args.gas_price_cap,
-        transaction_strategy: match args.transaction_strategy {
+    let submission_nodes = args
+        .transaction_submission_nodes
+        .into_iter()
+        .enumerate()
+        .map(|(index, url)| {
+            let transport = create_instrumented_transport(
+                HttpTransport::new(client.clone(), url, index.to_string()),
+                metrics.clone(),
+            );
+            web3::Web3::new(transport)
+        })
+        .collect::<Vec<_>>();
+    for node in &submission_nodes {
+        let node_network_id = node.net().version().await.unwrap();
```

Comment:
nit: How likely is this `unwrap` to fail? Seems odd to have an `unwrap` right before an assertion. Maybe they could be combined somehow. Ultimately it probably wouldn't make huge difference in case of a failed `unwrap`.

Reply:
Good point. This code is not new (I just moved it) but I guess the reasoning behind the unwrap is that it happens in main at the app startup so panicking is not that big of a deal and also enforcing the user to enter valid arguments.